### PR TITLE
fix(TDS-5962): fix not translated "Select All"

### PR DESCRIPTION
**Pull request by abdelillah-tech**
**What is the problem this PR is trying to solve?**
Not translated "Select all" in HeaderCheckBox component

**What is the chosen solution to this problem?**
Add translator function using useTranslate hook in the component

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: .changeset/four-tomatoes-develop.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@talend/react-components': minor
```
Comment (Nigui) [inline .changeset/four-tomatoes-develop.md:2]: isn't a patch ?

**Review by Nigui — APPROVED** [review]

File: packages/components/src/VirtualizedList/HeaderCheckbox/HeaderCheckbox.component.js [hunk 2]
```diff
@@ -3,30 +3,31 @@ import React, { useMemo } from 'react';
...
 /**
  * Header renderer that displays a "Select All" checkbox on header
  */
-function HeaderCheckbox({ columnData, t }) {
+function HeaderCheckbox({ columnData }) {
 	if (!columnData.onToggleAll) {
 		return null;
 	}
 
+	const { t } = useTranslation(I18N_DOMAIN_COMPONENTS);
```
Comment (sgangnant) [inline packages/components/src/VirtualizedList/HeaderCheckbox/HeaderCheckbox.component.js:17]: Please check the error below
Hooks are supposed to be the first instructions of a function component (aside from assignations)

Comment (abdelillah-tech) [reply packages/components/src/VirtualizedList/HeaderCheckbox/HeaderCheckbox.component.js:17]: it is the case for useMemo too

**Review by smouillour — APPROVED** [review]
LGTM

**Pull request merged**